streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -364,6 +368,62 @@ public void setUncaughtExceptionHandler(final Thread.UncaughtExceptionHandler eh
         }
     }
 
+    /**
+     * Set the handler invoked when a {@link StreamsConfig#NUM_STREAM_THREADS_CONFIG internal thread} abruptly
+     * terminates due to an uncaught exception.
+     *
+     * @param eh the uncaught exception handler of type {@link StreamsUncaughtExceptionHandler} for all internal threads; {@code null} deletes the current handler
+     * @throws IllegalStateException if this {@code KafkaStreams} instance is not in state {@link State#CREATED CREATED}.
+     */
+    public void setUncaughtExceptionHandler(final StreamsUncaughtExceptionHandler eh) {
```

Comment:
```suggestion
    public void setUncaughtExceptionHandler(final StreamsUncaughtExceptionHandler uncaughtExceptionHandler) {
```

We prefer to resist the urge to abbreviate, especially in the public-facing APIs.

Reply:
should I also change the old handler? it uses the same name

Reply:
Oh, sure. Now I know why you picked this name :)